File: javascript/Basic-Algorithm-Scripting/chunkyMonkey.js
/*

Write a function that splits an array (first argument) into groups the length of size (second argument) and returns them as a two-dimensional array.

*/


function chunkArrayInGroups(arr, size) {
    let ret = [];
    for(let i=0; i<arr.length; i+=size)
        ret.push(arr.slice(i, i+size));
    return ret
}
  
chunkArrayInGroups(["a", "b", "c", "d"], 2);

File: swift/chunkyMonkey.swift
/*

Write a function that splits an array (first argument) into groups the length of size (second argument) and returns them as a two-dimensional array.

*/

func chunkArrayInGroups(_ arr:[String], _ chunckSize:Int) -> [[String]] {
    var ret: [[String]] = []
    var ix = 0
    var maxIx = 0
    while ix < arr.count {
        maxIx = ix+chunckSize < arr.count ? ix+chunckSize : arr.count
        ret.append( Array(arr[ix..<maxIx]) )
        ix += chunckSize
    }
    return ret
}



print(chunkArrayInGroups(["ab", "cd", "ef", "gh", "ij", "kl", "derp"], 2))
